cli/tools/coverage.rs

```diff
@@ -171,392 +172,338 @@ impl CoverageCollector {
+      // if the basic block containing the branch was never executed or a number indicating how
+      // often that branch was taken.
+      //
+      // However with the data we get from v8 coverage profiles it seems we can't actually hit
+      // this as appears it won't consider any nested branches it hasn't seen but its here for
+      // the sake of accuracy.
+      let taken = if block_hits > 0 {
+        Some(range.count)
+      } else {
+        None
+      };
 
-      if execution_count != 0 {
-        functions_hit += 1;
-      }
+      coverage_report.branches.push(BranchCoverageItem {
+        line_index,
+        block_number,
+        branch_number,
+        taken,
+        is_hit: range.count > 0,
+      })
     }
+  }
 
-    println!("FNF:{}", functions_found);
-    println!("FNH:{}", functions_hit);
-
-    let mut branches_found = 0;
-    let mut branches_hit = 0;
-    for (block_number, function) in script_coverage.functions.iter().enumerate()
-    {
-      let block_hits = function.ranges[0].count;
-      for (branch_number, range) in function.ranges[1..].iter().enumerate() {
-        let source_line = script_source
-          .chars()
-          .take(range.start_offset)
-          .filter(|c| *c == '\n')
-          .count()
-          + 1;
-
-        let line_index = if let Some(source_map) = maybe_source_map.as_ref() {
-          source_map
-            .tokens()
-            .find(|token| token.get_dst_line() as usize == source_line)
-            .map(|token| token.get_src_line() as usize)
-            .unwrap_or(0)
-        } else {
-          source_line
-        };
-
-        // From https://manpages.debian.org/unstable/lcov/geninfo.1.en.html:
-        //
-        // Block number and branch number are gcc internal IDs for the branch. Taken is either '-'
-        // if the basic block containing the branch was never executed or a number indicating how
-        // often that branch was taken.
-        //
-        // However with the data we get from v8 coverage profiles it seems we can't actually hit
-        // this as appears it won't consider any nested branches it hasn't seen but its here for
-        // the sake of accuracy.
-        let taken = if block_hits > 0 {
-          range.count.to_string()
-        } else {
-          "-".to_string()
-        };
-
-        println!(
-          "BRDA:{},{},{},{}",
-          line_index + 1,
-          block_number,
-          branch_number,
-          taken
-        );
-
-        branches_found += 1;
-        if range.count > 0 {
-          branches_hit += 1;
+  // TODO(caspervonb): collect uncovered ranges on the lines so that we can highlight specific
+  // parts of a line in color (word diff style) instead of the entire line.
+  let mut line_counts = Vec::with_capacity(text_lines.lines_count());
+  for line_index in 0..text_lines.lines_count() {
+    let line_start_offset = text_lines.line_start(line_index);
+    let line_end_offset = text_lines.line_end(line_index);
```

Comment:
It might be good to improve `text_lines` to return an iterator of line information, but this is an `O(1)` lookup so not critical... just would be nicer.